### ISSUE TITLE
Add support for burning in text-based subtitles

### ISSUE THREAD

**Issue by samhutchins** [donmelton/other_video_transcoding#76]
`ffmpeg` has a `subtitle` filter for burning in text-based subtitles: https://ffmpeg.org/ffmpeg-filters.html#subtitles-1

I'm not asking for external subtitle support, but if the input file has a text-based subtitle track muxed in already it'd be awesome to be able to burn it into the video stream, just like the image-based subtitles

**Comment by donmelton** [comment]
@samhutchins Good idea. And, yeah, internal subtitles are likely _much_ easier to implement.

**Comment by TheClaus** [comment]
Was coming here to ask this and glad to see it on here.  I have some media that is using text based subtitles and would love to burn it in like we can with image based subtitles.  Makes my wife happy when subtitles just magically show up when they are suppose to.

**Comment by donmelton** [comment]
@samhutchins I've been looking at the `subtitles` filter as well as the `ass` filter today and it seems this isn't as easy as I thought.

To make the `subtitles` filter work reliably with a embedded subtitle, you really need to export that _single_ subtitle to a separate file. This is because it's not possible to choose from multiple subtitles that could be embedded within the original source file.

As for the `ass` filter (great name, BTW), it requires subtitles in the ASS format so you have to convert them first. Which means, again, you have to export the embedded subtitle to a separate file. And that's likely since most text-based subtitles are in SRT format.

So I'm going to put this on the shelf for now until I decide it's worth it to write all the scaffolding to manage and manipulate a temporary file to contain the exported subtitle. And it's a lot of scaffolding due to the various error conditions.
Sorry about that.

**Comment by DanielEgelseer** [comment]
@donmelton 
Please shift some priority towards it.

I played around with this tool for a bit and i really like how the files look afterwards!
(espessially since it supports 10bit nvenc)
BUT i would mainly use it to convert anime and there the files ALL come with ass subtitles integrated...
And i would need to burn these in (for playback reasons...).

Not supporting ass makes this tool useless to me sadly T-T

If you need test-files or a tester i would be happy to help since this tool would help me greatly!

**Comment by donmelton** [comment]
@DanielEgelseer Thanks for the feedback, sir. 👍
